src/mono/mono/mini/interp/transform.c

```diff
@@ -8669,16 +8867,25 @@ generate (MonoMethod *method, MonoMethodHeader *header, InterpMethod *rtm, MonoG
 	if (td->has_localloc)
 		interp_fix_localloc_ret (td);
 
-	if (td->verbose_level)
+	if (td->verbose_level) {
+		g_print ("\nUnoptimized IR:\n");
 		mono_interp_print_td_code (td);
+	}
 
 	if (td->optimized) {
-		interp_optimize_code (td);
+		MONO_TIME_TRACK (mono_interp_stats.optimize_time, interp_optimize_code (td));
 		interp_alloc_offsets (td);
+		// Offset allocator uses computed ref counts from var values. We have to free this
+		// table later here.
+		if (td->var_values != NULL) {
+			g_free (td->var_values);
+			td->var_values = NULL;
+		}
 #if HOST_BROWSER
 		if (mono_interp_opt & INTERP_OPT_JITERPRETER)
 			jiterp_insert_entry_points (rtm, td);
 #endif
+		interp_squash_initlocals (td);
```

Comment:
it would be better to do this before inserting jiterpreter entry points instead of after, i think